extensions/resource-deployment/src/ui/deploymentDialog.ts

```diff
@@ -55,74 +73,119 @@ export class DeploymentDialog extends DialogBase {
 		}).component();
 		this.variables[fieldInfo.variableName] = defaultValue;
-		this._toDispose.push(component.onTextChanged(() => { this.variables[fieldInfo.variableName] = component.value; }));
-		fields.push({ title: fieldInfo.label, component: component });
+		this._toDispose.push(input.onTextChanged(() => { this.variables[fieldInfo.variableName] = input.value; }));
+		return input;
 
-		if (fieldInfo.type === FieldType.Password && fieldInfo.confirmationRequired) {
-			const confirmPasswordComponent = view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({ ariaLabel: fieldInfo.confirmationLabel, inputType: inputType, required: true }).component();
-			fields.push({ title: fieldInfo.confirmationLabel, component: confirmPasswordComponent });
+	}
 
-			this._dialogObject.registerCloseValidator((): boolean => {
-				const passwordMatches = component.value === confirmPasswordComponent.value;
-				if (!passwordMatches) {
-					this._dialogObject.message = { level: azdata.window.MessageLevel.Error, text: localize('passwordNotMatch', "{0} doesn't match the confirmation password", fieldInfo.label) };
+	private createNumberField(view: azdata.ModelView, fieldInfo: DialogFieldInfo): azdata.InputBoxComponent {
+		const input = view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
+			value: fieldInfo.defaultValue, ariaLabel: fieldInfo.label, inputType: 'number', min: fieldInfo.min, max: fieldInfo.max, required: fieldInfo.required
+		}).component();
+		this._toDispose.push(input.onTextChanged(() => { this.variables[fieldInfo.variableName] = input.value; }));
+		return input;
+	}
+
+	private createTextField(view: azdata.ModelView, fieldInfo: DialogFieldInfo): azdata.InputBoxComponent {
+		const input = view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
+			value: fieldInfo.defaultValue, ariaLabel: fieldInfo.label, inputType: 'text', min: fieldInfo.min, max: fieldInfo.max, required: fieldInfo.required, placeHolder: fieldInfo.placeHolder
+		}).component();
+		this._toDispose.push(input.onTextChanged(() => { this.variables[fieldInfo.variableName] = input.value; }));
+		return input;
+	}
+
+	private createPasswordField(view: azdata.ModelView, fieldInfo: DialogFieldInfo): { title: string, component: azdata.Component }[] {
+		const components: { title: string, component: azdata.Component }[] = [];
+		const passwordInput = view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
+			ariaLabel: fieldInfo.label, inputType: 'password', required: fieldInfo.required, placeHolder: fieldInfo.placeHolder
+		}).component();
+		this._toDispose.push(passwordInput.onTextChanged(() => { this.variables[fieldInfo.variableName] = passwordInput.value; }));
+		components.push({ title: fieldInfo.label, component: passwordInput });
+
+		if (fieldInfo.type === FieldType.SQLPassword) {
```

Comment:
in all of these create* methods, just to be safe, we should do some input validation to ensure that they’re not null/undefined

Reply:
next step, I am going to move this dialog (rename it properly) to notebook extension, so that it can be reused by any scenarios that we need to do UI assisted notebooks. i will add validation when doing that.